src/app/api/update-profile/route.ts

```
@@ -0,0 +1,51 @@
+  // If no token → return Unauthorized response
+  if (!token?.accessToken) {
+    return NextResponse.json({ message: "Unauthorized" }, { status: 401 });
+  }
+
+  // =======================================================
+  // Step 3: Filter ONLY the fields allowed to be updated
+  // This protects backend from receiving unwanted fields
+  // like _id, photo, role, createdAt, wishlist, addresses
+  // =======================================================
+  const allowedFields = ["firstName", "lastName", "email", "phone"];
+  const filteredBody: any = {};
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)